### DISCUSSION TITLE
HMR in ViteNET.MVC example

### DISCUSSION THREAD

**Discussion by martinmikulka** [Eptagone/Vite.AspNetCore#47] [Q&A]
Hello, at the beginning, thanks for the effort you put into this great package.

While examining of the ViteNET.MVC example, I have noticed that the Hot Module Replacement (HMR) does not work. Or to be precise, it works, but in some weird way. 

When the module is changed, a complete page reload occurs in the browser. The state of the counter is reset after the reload. 
But it is expected, that HMR will reload only the changed module while retaining the counter's state.

I have been trying to figure it out, but no luck yet. So at least I am sharing this information, maybe some more experienced developer will make it working.

Thx.

**Comment by jessedijkstra** [comment]
Hey,

I've been struggling with this as well. For me the key was to add this for React:
```
    <environment exclude="Production">
        <script type="module">
            import RefreshRuntime from "http://localhost:5173/@@react-refresh"
            RefreshRuntime.injectIntoGlobalHook(window)
            window.$RefreshReg$ = () => { }
            window.$RefreshSig$ = () => (type) => type
            window.__vite_plugin_react_preamble_installed__ = true
        </script>
    </environment>
```
I'll be working on creating a PR that injects the react-preamble with an option to include this.
